.github/workflows/e2e-tests.yml

```diff
@@ -8,32 +8,20 @@ on:
 
 jobs:
 
-  docker-build-e2e:
-    runs-on: ubuntu-20.04
-    steps:
-      - name: Delete unnecessary tools folder to prevent "out of space" error
-        run: rm -rf /opt/hostedtoolcache
-
-      - name: Checkout
-        uses: actions/checkout@v4
-
-      - name: Build E2E Docker Image
-        uses: ./.github/actions/docker-build-e2e
-        with:
-          name: oisy-e2e-image
-
   e2e:
     runs-on: ubuntu-latest
-    needs: docker-build-e2e
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
       - name: Download E2E Docker image artifact
-        uses: actions/download-artifact@v4
+        id: download-artifact
+        uses: dawidd6/action-download-artifact@v6
```

Comment:
the difference between `actions/download-artifact@v4` and `dawidd6/action-download-artifact@v6`, for what I understand [in the docs](https://github.com/dawidd6/action-download-artifact), is the possibility to pass an uploaded artifacts from a workflow to the other, correct? isn't there any "official" workflow to rely upon?

I would be careful with using "unofficial" packages/workflows; I will remit to @bitdivine and @peterpeterparker regarding this; I am not really sure what is our policy here, since it is the first time for me too

Reply:
Yes, `dawidd6/action-download-artifact@v6` allows to download artifacts created in separate workflows (with `actions/download-artifact@v4` you can only do it within a single workflow run). Unfortunately, there is no official action for that, GitHub suggests to use its REST API to do so. So, in the end `dawidd6/action-download-artifact@v6` is just a wrapper around official API endpoint.

We already use at least one unofficial action (https://github.com/dfinity/oisy-wallet/blob/ci(e2e)/separate-e2e-workflows/.github/actions/create-pr/action.yml#L23), but in case you have doubts in `dawidd6/action-download-artifact@v6` please let me know.

Reply:
I sincerely have not strong opinion on using it or not, so, I would wait for Max and David to give their input

Reply:
Note that standard caches are valid across workflows and across different runs.  That is why [the binding check](https://github.com/dfinity/oisy-wallet/actions/workflows/binding-checks.yml) usually completes in about 1m20s for PRs that have only frontend changes.

I would recommend making a GitHub action in `.github/actions/oisy-backend` that does nothing but provide the Oisy backend Wasm, ideally from a cached build but building it if necessary.  The runtime of that action will be strictly less than the binding check, so less than 1m20s for frontend-only PRs and about 4-5m for a typical backend change.  That action can be used in many places, for example in the e2e tests, the binding tests and elsewhere.  You can use whatever tools you want to build that action; you don't need custom tools but if you  find them super-performant and the implementation looks secure, go ahead!

I would recommend doing the same for the frontend.  So, making an action that provides the `/build/` directory populated with all the frontend assets.  This action would have to take network as an argument.  This way the frontend doesn't need to be rebuilt if there are no frontend changes.

For the e2e tests themselves, note that you can request a bigger runner with more CPU cores, so you can run more tests in parallel.  Alternatively, if that is likely to cause concurrency issues, you can manually split the e2e tests into shards, [as we do for the nns-dapp](https://github.com/dfinity/nns-dapp/blob/main/.github/workflows/build.yml#L77).

Reply:
Thanks for the very useful insights @bitdivine. Your suggestion does sound good, I'm already working on implementing it. Related PRs will follow. 

For now, closing this PR.